src/main/java/org/broadinstitute/hellbender/tools/spark/validation/CompareDuplicatesSpark.java

```diff
@@ -148,19 +156,66 @@ protected void runTool(final JavaSparkContext ctx) {
                 ReadUtils.getLibrary(read, bHeader.getValue())), read));
         JavaPairRDD<Integer, Tuple2<Iterable<GATKRead>, Iterable<GATKRead>>> cogroup = firstKeyed.cogroup(secondKeyed, getRecommendedNumReducers());
 
+        JavaRDD<Tuple2<Iterable<GATKRead>, Iterable<GATKRead>>> subsettedByStart = cogroup.flatMap(v1 -> {
+            List<Tuple2<Iterable<GATKRead>, Iterable<GATKRead>>> out = new ArrayList<>();
+
+            Iterable<GATKRead> iFirstReads = v1._2()._1();
+            Iterable<GATKRead> iSecondReads = v1._2()._2();
+
+            Map<Integer, List<GATKRead>> firstReadsMap = splitByStart(iFirstReads);
+            Map<Integer, List<GATKRead>> secondReadsMap = splitByStart(iSecondReads);
+
+            for (Integer i : firstReadsMap.keySet()) {
+                out.add(new Tuple2<>(firstReadsMap.get(i), secondReadsMap.get(i)));
+            }
+            return out.iterator();
+        });
+
+        if (output!=null && output2!=null) {
+
+            JavaRDD<Tuple2<Iterable<GATKRead>, Iterable<GATKRead>>> unequalGroups = subsettedByStart.filter(v1 -> {
```

Comment:
pull this out into a function for clarity

Reply:
Oh, heh, I meant the part in the if statement that writes the output, but that's actually useful too.